openerp/service/server.py

```diff
@@ -232,7 +232,12 @@ def close_socket(self, sock):
         sock.close()
 
 class ThreadedServer(CommonServer):
+    initial_threads = []
+    
     def __init__(self, app):
+        for t in threading.enumerate():
+            self.initial_threads.append(t)
```

Comment:
When ran normally, yes, the main thread is the only thread at this point, but in the specific case where odoo unit tests are run with pytest in pydev, there is an additional one and odoo should not wait for it. This additional thread is not spawned by Odoo, but rather by either pytest or pydev.